src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/IReflectDataflow.cs

```diff
@@ -164,6 +164,9 @@ class TestType
 			}
 
 			[Kept]
+			// MyReflectOverType is not intrinsically understood by the analysis, so
+			// it doesn't satisfy the PublicFields | NonPublicFields requirement.
+			[ExpectedWarning ("IL2075")]
```

Comment:
Looks like this test was added in https://github.com/dotnet/linker/pull/2831 to cover some real world scenario. This happens to be trim safe even though the analysis doesn't see it (the warning is correct). Would it make sense to introduce one more test where this leads to an actual bug?

...just in case we do something for the real world scenario later to make this not warn (it does feel unlikely we'd do it but still) and delete the ExpectedWarning. We'd accidentally lose test coverage for this PR since this is essentially the only regression test for "dataflow shouldn't return empty for newobj".